Scripts DICOM/writeTROTSToDICOM.py

```diff
@@ -215,6 +224,204 @@
+                    beaminfo["ControlPoints"].append(copy.deepcopy(controlpointinfo))
+                    currentbeamlist.append(copy.deepcopy(beaminfo))
+                    beaminfo["BeamNumber"] = beaminfo["BeamNumber"] + 1
+                    beaminfo["FileBeamNumber"] = int(row[0])
+                    beaminfo["RangeShifter"] = row[4]
+                    beaminfo["FinalCumulativeMetersetWeight"] = 0
+                    beaminfo["ControlPoints"] = []
+                    controlpointinfo["CumulativeMetersetWeight"] = 0
+                    controlpointinfo["ControlPointNumber"] += 1
+                    controlpointinfo["BeamEnergy"] = row[1]
+                    controlpointinfo["MetersetWeights"] = [mat["solutionX"][rowindex]]
+                    controlpointinfo["ScanSpotPositions"] = beamlistfolder["BeamList"][int(patientFolder[-2:])-1][rowindex][2:4].tolist()
+            beaminfo["FinalCumulativeMetersetWeight"] = controlpointinfo["CumulativeMetersetWeight"] + sum(controlpointinfo["MetersetWeights"])
+            beaminfo["ControlPoints"].append(copy.deepcopy(controlpointinfo))
+            currentbeamlist.append(beaminfo) 
+            
+            rtds.FractionGroupSequence = Sequence()
+            fractionds = Dataset()
+            fractionds.FractionGroupNumber = 1
+            fractionds.NumberOfFractionsPlanned = 1
+            fractionds.NumberOfBeams = len(currentbeamlist)
+            fractionds.NumberOfBrachyApplicationSetups = 0
+            fractionds.BeamSequence = Sequence()
+            for beaminfo in currentbeamlist:
+                be = Dataset()
+                be.BeamMeterset = beaminfo["FinalCumulativeMetersetWeight"]
+                be.BeamNumber = beaminfo["BeamNumber"]
+                fractionds.BeamSequence.append(be)
+            rtds.FractionGroupSequence.append(fractionds)
+            
+            for beaminfo in currentbeamlist:
+                be = Dataset()
+                be.Manufacturer = ""
+                be.InstitutionName  = ""
+                be.ManufacturerModelName  = ""
+                be.TreatmentMachineName   = ""
+                be.InstitutionAddress = ""
+                be.PrimaryDosimeterUnit = "MU"
+                be.BeamNumber             = beaminfo["BeamNumber"]
+                be.BeamName               = ""
+                be.BeamType               = 'STATIC'
+                be.RadiationType          = "PROTON"
+                be.TreatmentDeliveryType  = 'TREATMENT'
+                be.NumberOfWedges         = 0
+                be.NumberOfCompensators   = 0
+                be.NumberOfBoli           = 0
+                be.NumberOfBlocks         = 0
+                be.FinalCumulativeMetersetWeight = beaminfo["FinalCumulativeMetersetWeight"]
+                be.ScanMode                   = 'MODULATED'
+                be.SourceAxisDistance = SAD

```

Comment:
Should be rather: https://dicom.innolitics.com/ciods/rt-ion-plan/rt-ion-beams/300a03a2/300a030a